pkg/entity/register/worker.go

```diff
@@ -0,0 +1,111 @@
+package register
+
+import (
+	"context"
+	"time"
+
+	"github.com/newrelic/infrastructure-agent/internal/agent/id"
+	"github.com/newrelic/infrastructure-agent/pkg/backend/identityapi"
+	"github.com/newrelic/infrastructure-agent/pkg/entity"
+	"github.com/newrelic/infrastructure-agent/pkg/fwrequest"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
+)
+
+type worker struct {
+	agentIDProvide      id.Provide
+	client              identityapi.RegisterClient
+	reqsToRegisterQueue <-chan fwrequest.EntityFwRequest
+	reqsRegisteredQueue chan<- fwrequest.EntityFwRequest
+	maxBatchSize        int
+	maxBatchDuration    time.Duration
+}
+
+func NewWorker(
+	agentIDProvide id.Provide,
+	client identityapi.RegisterClient,
+	reqsToRegisterQueue <-chan fwrequest.EntityFwRequest,
+	reqsRegisteredQueue chan<- fwrequest.EntityFwRequest,
+	maxBatchSize int,
+	maxBatchDuration time.Duration,
+) *worker {
+	return &worker{
+		agentIDProvide:      agentIDProvide,
+		client:              client,
+		reqsToRegisterQueue: reqsToRegisterQueue,
+		reqsRegisteredQueue: reqsRegisteredQueue,
+		maxBatchSize:        maxBatchSize,
+		maxBatchDuration:    maxBatchDuration,
+	}
+}
+
+func (w *worker) Run(ctx context.Context) {
+	timer := time.NewTimer(w.maxBatchDuration)
+
+	// data for register batch call
+	batch := make(map[entity.Key]fwrequest.EntityFwRequest, w.maxBatchSize)
+	batchSize := 0
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+
+		case req := <-w.reqsToRegisterQueue:
+			batchSize++
+
+			// TODO update when entity key retrieval is fixed
+			eKey := entity.Key(req.Data.Entity.Name)
+			batch[eKey] = req
+
+			// TODO add 1MB payload size platform limitation
+
+			// TODO add timer trigger
```

Comment:
Is this the todo for timer written bellow or a different timer?

Reply:
nope, removed